src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs

```diff
@@ -88,15 +164,23 @@ public void ValidateDisplayNames(string inputExpression, string outputExpression
             var result = _engine.Check(inputExpression, r1);
             Assert.True(result.IsSuccess);
 
-            if (toDisplay)
-            {
-                var outDisplayExpression = _engine.GetDisplayExpression(inputExpression, r1);
-                Assert.Equal(outputExpression, outDisplayExpression);
-            }
-            else
+            // Below Record r2 Tests the second method where we provide DisplayNameProvider via constructor to 
+            // initialize the DisplayNameProvider for derived record types.
+            var r2 = new LazyRecordType();
+
+            var records = new RecordType[] { r1, r2 };
```

Comment:
It'd be good to also replicate the Check call with r1 and r2.

Reply:
Would catch that the type definitions aren't actually the same.

Reply:
Moved the Check call into the loop.